packages/generator/src/utils/fetch-latest-version-for.ts

```diff
@@ -26,6 +26,8 @@ export const fetchLatestVersionsFor = async <T extends Record<string, string>>(
 
   return {
     isFallback: fallbackUsed,
-    value: Object.fromEntries(updated),
+    value: updated.reduce((object, [key, value]) => {
+      return Object.assign(object, {[key]: value})
```

Comment:
can we add `{}` as the first argument to `Object.assign` so we're updating immutably rather than mutating the object every iteration?